test/integration/edition_edit_test.rb

```diff
@@ -108,53 +94,149 @@ class EditionEditTest < IntegrationTest
-      should "show 'Unpublish' header and 'Continue' button" do
-        within :css, ".gem-c-heading" do
-          assert page.has_text?("Unpublish")
-        end
-        assert page.has_button?("Continue")
+      should "not show unpublish tab when user is not govuk editor" do
+        assert page.has_no_text?("Unpublish")
       end
+    end
 
-      should "show 'cannot be undone' banner" do
-        assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+    context "has required permissions" do
+      setup do
+        login_as(FactoryBot.create(:user, :govuk_editor, name: "Stub User"))
+        edition = FactoryBot.create(:guide_edition, state: "draft")
+        visit edition_path(edition)
       end
 
-      should "show 'Redirect to URL' text, input box and example text" do
-        assert page.has_text?("Redirect to URL")
-        assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
-        assert page.has_css?(".govuk-input", count: 1)
+      context "when state is publish" do
+        setup do
+          @edition = FactoryBot.create(:guide_edition, state: "published")
+          visit edition_path(@edition)
+          click_link("Unpublish")
+        end
+
+        should "show 'Unpublish' header and 'Continue' button" do
+          within :css, ".gem-c-heading" do
+            assert page.has_text?("Unpublish")
+          end
+          assert page.has_button?("Continue")
+        end
+
+        should "show 'cannot be undone' banner" do
+          assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+        end
+
+        should "show 'Redirect to URL' text, input box and example text" do
+          assert page.has_text?("Redirect to URL")
+          assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
+          assert page.has_css?(".govuk-input", count: 1)
+        end
+
+        should "navigate to 'confirm-unpublish' page when clicked on 'Continue' button" do
+          click_button("Continue")
+          assert_equal(page.current_path, "/editions/#{@edition.id}/unpublish/confirm-unpublish")
+        end
       end
 
-      should "navigate to 'confirm-unpublish' page when clicked on 'Continue' button" do
-        click_button("Continue")
-        assert_equal(page.current_path, "/editions/#{@edition.id}/unpublish/confirm-unpublish")
+      context "when state is not publish" do
+        setup do
+          edition = FactoryBot.create(:guide_edition, state: "draft")
+          visit edition_path(edition)
+        end
+
+        should "not show unpublish tab" do
+          assert page.has_no_text?("Unpublish")
+        end
       end
     end
+  end
 
-    context "confirm unpublish" do
+  context "admin tab" do
+    context "do not have required permissions" do
       setup do
-        click_link("Unpublish")
-        click_button("Continue")
+        user = FactoryBot.create(:user, name: "Stub User")
```

Comment:
I think we should move the code from this setup down into the only test that is using it (to match the other test for welsh editors).

Reply:
I now have refactored the test file and think refactor should fix this.